runelite-client/src/main/java/net/runelite/client/RuneLite.java

```diff
@@ -373,38 +375,50 @@ public void shutdown()
 		discordService.close();
 	}
 
-	private static class FileConverter implements ValueConverter<File>
+	@VisibleForTesting
+	public static void setInjector(Injector injector)
+	{
+		RuneLite.injector = injector;
+	}
+
+	private static class ConfigFileConverter implements ValueConverter<File>
 	{
 		@Override
 		public File convert(String fileName)
 		{
-			Path path = Paths.get(fileName);
-			if (path.isAbsolute()
+			final File file;
+
+			if (Paths.get(fileName).isAbsolute()
 				|| fileName.startsWith("./")
 				|| fileName.startsWith(".\\"))
 			{
-				return new File(fileName);
+				file = new File(fileName);
 			}
 			else
 			{
-				return new File(RuneLite.RUNELITE_DIR, fileName);
+				file = new File(RuneLite.RUNELITE_DIR, fileName);
 			}
+
+			final Path path = Paths.get(file.getAbsolutePath());
+
+			if (Files.exists(path) && (!Files.isRegularFile(path) || !Files.isWritable(path)))
```

Comment:
Why do we convert to nio just for this?
```suggestion
			if (file.exists() && (!file.isFile() || !file.canWrite()))
```

Reply:
Because I was trying file.isWrite and it wasnt there, lol. Updated